docs/plugins/testing_workshop_docs/3-readers-and-fixtures.md

```diff
@@ -0,0 +1,151 @@
+# 3: Readers and fixtures  
+
+This lesson explains how to use and test a plugin's reader function, built-in fixtures, custom fixtures, and enclosed testing. 
+
+## Other lessons in this tutorial:  
+  
+* 1: [Python’s assert keyword](./1-pythons-assert-keyword.md) 
+* 2: [Pytest testing framework](./2-pytest-testing-frameworks.md)  
+* 3: This lesson (Readers and fixtures)  
+* 4: [Test coverage](./4-test-coverage.md)  
+* Resource links: [Testing resources](./testing-resources.md)  
+
+### This lesson covers:   
+* [Readers](#reader)  
+* [Built-in fixtures](#built-in-fixtures)  
+* [Custom fixtures and round-trip tests](#custom-fixtures-and-round-trip-tests)  
+* [Enclosed testing](#enclosed-testing)  
+  
+### Resources  
+The example plugin and all the tests discussed in this lesson are available in [this GitHub repository](https://github.com/DragaDoncila/plugin-tests).  
+  
+## Introduction  
+In this lesson, we discuss a plugin called [plugin_tests](https://github.com/DragaDoncila/plugin-tests/tree/main/src/plugin_tests), generated using the [cookiecutter](https://github.com/napari/cookiecutter-napari-plugin), which has a reader and a widget. The reader is the cookiecutter numpy file reader, `napari_get_reader`. It checks whether a path ends in `.npy`. If it doesn't, it returns `None`, and if it does, it returns the `reader_function`, which loads the data. 
```

Comment:
To reduce the chances of someone confusing a pytest plugin with a napari plugin (both have this functionality) I suggest making this explicit. I would also be a little more specific with the plugin to reduce ambiguity:

```suggestion
In this lesson, we discuss a napari plugin called [plugin_tests](https://github.com/DragaDoncila/plugin-tests/tree/main/src/plugin_tests), generated using the [cookiecutter](https://github.com/napari/cookiecutter-napari-plugin), which has a reader and a widget. The reader is the cookiecutter [NumPy `.npy` file](https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html#npy-format) reader, `napari_get_reader`. It checks whether a path ends in `.npy`. If it doesn't, it returns `None`, and if it does, it returns the `reader_function`, which loads the data. 
```

Reply:
Great suggestion!